### PR TITLE
Swifttool changes to modify current rings

### PR DESCRIPTION
**Pull request by ryshah**
Added functionality to set weight for existing device and removing the device from existing rings.

### CODE REVIEW

File: swifttool/client.py [hunk 1]
```diff
@@ -189,6 +244,14 @@ def generate_script(self, outdir, name='ring_builder.sh', rebalance=True,
         os.chmod(outfile, st.st_mode | stat.S_IEXEC)
         return outfile
 
+    def is_devpresent(self, ringtype, zone, node, port, disk):
+        command = self._ring_search_command(ringtype, zone, node,
+                                            port, disk)
+        with open(os.devnull, 'w') as devnull:
+            rc = subprocess.call(command, shell=True, stdout=devnull,
```
Comment (omgjlk) [inline swifttool/client.py]: Why do you need to devnull the stdout? You don't process it anyway, what's the harm in having the output happen?

**Comment by omgjlk** [comment]
I think this is fine, but I want @craigtracey to review as well.

File: swifttool/client.py (hunk 1 above)
Comment (ryshah) [inline swifttool/client.py]: @j2sol Ring search outputs device info on stdout. In past we had run into issue when subprocess would randomly terminate (in different project) with stdout buffer overflows. That probably wont be case here though... Do you want me to remove the devnull?

File: swifttool/client.py [hunk 2]
```diff
@@ -96,26 +97,51 @@ def __init__(self, data=None):
         self.replicas = 3
         self.min_part_hours = 1
         self.zones = {}
+        # Area to build new or update existing rings
```
Comment (craigtracey) [inline swifttool/client.py:100]: Please use **init** for initialization only.  We should not be moving files around the **init** method.

File: swifttool/client.py [hunk 3]
```diff
@@ -86,7 +87,7 @@ def get_disk_size_serial(ip, blockdev):
 
 class SwiftRingsDefinition(object):
 
-    def __init__(self, data=None):
+    def __init__(self, outdir, data=None):
```
Comment (craigtracey) [inline swifttool/client.py]: If we are passing outdir in, we should set self.outdir.

File: swifttool/client.py [hunk 4]
```diff
@@ -96,26 +97,51 @@ def __init__(self, data=None):
         self.replicas = 3
         self.min_part_hours = 1
         self.zones = {}
+        # Area to build new or update existing rings
+        self.workspace = tempfile.mkdtemp()
```
Comment (craigtracey) [inline swifttool/client.py:101]: It is possible that this is not cleaned up

File: swifttool/client.py [hunk 5]
```diff
@@ -154,32 +184,57 @@ def generate_commands(self, outdir, rebalance=True, meta=None):
                             blockdev = '/dev/%s' % match.group(1)
 
                             # treat size as weight and serial as metadata
-                            weight, serial = get_disk_size_serial(node, blockdev)
+                            weight, serial = get_disk_size_serial(node,
```
Comment (craigtracey) [inline swifttool/client.py:184]: Why is this changed?

File: swifttool/client.py [hunk 6]
```diff
@@ -189,6 +244,14 @@ def generate_script(self, outdir, name='ring_builder.sh', rebalance=True,
         os.chmod(outfile, st.st_mode | stat.S_IEXEC)
         return outfile
 
+    def is_devpresent(self, ringtype, zone, node, port, disk):
```
Comment (craigtracey) [inline swifttool/client.py]: This is a private method, so it should lead with an underscore.

File: swifttool/client.py [hunk 7]
```diff
@@ -29,6 +29,7 @@
 import sys
 import tempfile
 import yaml
+import glob
```
Comment (craigtracey) [inline swifttool/client.py]: nit: please alphabetize this

File: swifttool/client.py (hunk 1 above)
Comment (craigtracey) [inline swifttool/client.py]: Agree with @j2sol

File: swifttool/client.py (hunk 3 above)
Comment (ryshah) [inline swifttool/client.py]: Ok - i will remove outdir from **init** and also moving files logic

File: swifttool/client.py (hunk 5 above)
Comment (ryshah) [inline swifttool/client.py:184]: PEP8 complained... Line is 83 characters

File: swifttool/client.py (hunk 6 above)
Comment (ryshah) [inline swifttool/client.py]: Makes sense

File: swifttool/client.py (hunk 1 above)
Comment (ryshah) [inline swifttool/client.py]: Ok - will remove it

File: swifttool/client.py (hunk 4 above)
Comment (ryshah) [inline swifttool/client.py:101]: @craigtracey Cleanup still happens in bootstrap like before...During my testing it didn't leave temp directory... What do you suggest? **del** method?

**Comment by ryshah** [comment]
@craigtracey @j2sol Any other changes we need before this can be merged?

File: swifttool/client.py (hunk 4 above)
Comment (craigtracey) [inline swifttool/client.py:101]: Yes, this should be cleaned up in a destructor.

**Comment by craigtracey** [comment]
Let's get that one change and then we are good to go.  Thanks!

**Comment by ryshah** [comment]
Done. Thanks @craigtracey @j2sol

**Pull request merged**